Detectors/TOF/base/src/WindowFiller.cxx

```diff
@@ -124,8 +125,34 @@ void WindowFiller::fillOutputContainer(std::vector<Digit>& digits)
     int orbit_shift = mReadoutWindowData.size() / 3;
     int bc_shift = (mReadoutWindowData.size() % 3) * Geo::BC_IN_WINDOW;
     info.setBCData(mFirstIR.orbit + orbit_shift, mFirstIR.bc + bc_shift);
-    if (digits.size())
+    int firstPattern = mPatterns.size();
+    int npatterns = 0;
+
+    // check if patterns are in the current row
+    int ipatt = mCratePatterns.size() - 1;
+    for (std::vector<PatternData>::reverse_iterator it = mCratePatterns.rbegin(); it != mCratePatterns.rend(); ++it) {
+      if (it->row > mReadoutWindowCurrent)
+        break;
+
+      if (it->row < mReadoutWindowCurrent) { // this should not happen
+        LOG(ERROR) << "One pattern skipped because appears to occur early of the current row " << it->row << " < " << mReadoutWindowCurrent << " ?!";
+        mCratePatterns.erase(mCratePatterns.begin() + ipatt);
```

Comment:
``vector::erase`` is quite expensive (needs to reassign all elements after erased one), can't you simply invalidated the element?

Reply:
Since I always remove the last element I can also do a simple resize.
Do you think it is the same as you suggested?
Otherwise I can implement a procedure to invalidate it.

Reply:
I did not realize that it is always the last element. But then, it is better to use `pop_back()``

Reply:
done